src/kvm/types.rs

```diff
@@ -63,6 +63,32 @@ impl<'a> LaunchUpdateData<'a> {
     }
 }
 
+/// Inject a secret into the guest.
+#[repr(C)]
+pub struct LaunchSecret<'a> {
+    hdr_addr: u64,
+    hdr_len: u32,
+    guest_addr: u64,
+    guest_len: u32,
+    trans_addr: u64,
+    trans_len: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> LaunchSecret<'a> {
+    pub fn new(header: &'a Header, guest: usize, trans: &'a [u8]) -> Self {
+        Self {
+            hdr_addr: header as *const _ as _,
+            hdr_len: std::mem::size_of_val(header) as _,
```

Comment:
If you're using `size_of_val()` multiple times (I think you are), then `use ...`.